velox/functions/lib/tests/Re2FunctionsTest.cpp

```diff
@@ -854,5 +855,62 @@ TEST_F(Re2FunctionsTest, regexExtractAllBadArgs) {
   EXPECT_THROW(eval("123", "[a-z]+", 1), VeloxException);
 }
 
+TEST_F(Re2FunctionsTest, tryException) {
+  // Assert we throw without try.
+  VELOX_ASSERT_THROW(
+      evaluateOnce<std::string>(
+          "re2_extract(c0, c1)",
+          std::optional("123"),
+          std::optional("V)%(&r_b2o&Xw")),
+      "invalid regular expression:unexpected ): V)%(&r_b2o&Xw");
+
+  // Assert we can handle try's safely.
+  auto patternVector = makeConstant("V)%(&r_b2o&Xw", 3);
+  auto stringVector = makeFlatVector<std::string>({"abc", "mno", "pqr"});
+  auto input = makeRowVector({stringVector, patternVector});
+
+  auto result =
+      evaluate<SimpleVector<StringView>>("try(re2_extract(c0, c1))", input);
+
+  auto expected = makeNullConstant(TypeKind::VARCHAR, 3);
+  assertEqualVectors(expected, result);
+
+  // Try the same with re2match.
+  auto expectedMatch = makeNullConstant(TypeKind::BOOLEAN, 3);
+  auto resultMatch =
+      evaluate<SimpleVector<bool>>("try(re2_match(c0, c1))", input);
+  assertEqualVectors(expectedMatch, resultMatch);
+
+  // Try the same with re2search.
+  auto expectedSearch = makeNullConstant(TypeKind::BOOLEAN, 3);
+  auto resultSearch =
+      evaluate<SimpleVector<bool>>("try(re2_search(c0, c1))", input);
+  assertEqualVectors(expectedSearch, resultSearch);
+
+  // Try the same with like.
```

Comment:
This comment doesn't match the code below. I'd expect `try(like(c0, c1))` expression returning all nulls.

Reply:
Got it. Thank you for explaining. Please, revise the comment for clarity.